src/x/os/interrupt.go

```diff
@@ -50,6 +69,33 @@ func (i *InterruptError) Error() string {
 	return i.interrupt
 }
 
+// WatchForInterrupt watches for interrupts in a non-blocking fashion and closes
+// the interrupted channel when an interrupt is found. Use this method to ensure
+// child goroutines get closed properly by passing them the interrupted channel.
+// If the interrupted channel is closed, then the goroutine knows to stop
+// its work.
+func WatchForInterrupt(logger *zap.Logger, opts InterruptOptions) func() {
```

Comment:
I'd suggest merging these two use cases: async and blocking with an interface.

type InterruptWatcher interface {
  AsyncStart()
  AsyncStop()
  Wait() 
}

I don't fully understand why we need to this async functionality, but I think once you build this out it will be clearer.

Reply:
actually i get now, but yeah still want this as a single interface.

Reply:
We need the async functionality to allow the server to continue to startup while also watching for interrupts. Will also update the comment to make this more clear. Using an interface vs a la carte methods for interrupt watching feels more like a matter of personal preference so I'm inclined to leave as-is.

Reply:
yeah I dunno just feels like they are both related to each other so keeping them in one interface would be nicer, but I'll let it go